pkgs/applications/science/electronics/horizon-eda/default.nix

```diff
@@ -1,75 +1,38 @@
 { stdenv
 , boost
+, callPackage
 , coreutils
-, cppzmq
-, curl
-, libepoxy
-, fetchFromGitHub
-, glm
-, gtkmm3
-, lib
-, libarchive
-, libgit2
-, librsvg
 , libspnav
-, libuuid
-, opencascade-occt
-, pkg-config
-, podofo
 , python3
-, sqlite
 , wrapGAppsHook
 }:
 
+let
+  base = callPackage ./base.nix { };
+in
 stdenv.mkDerivation rec {
-  pname = "horizon-eda";
-  version = "2.5.0";
+  inherit (base) pname version src meta CASROOT;
```

Comment:
Could you just add a flag `enablePython ? false` and then unify the two packages under that flag in the same file? Then you can just

```nix
  horizon-eda = toPythonModule (pkgs.horizon-eda.override {
    enablePython = true;
    python3 = python;
  });
```

in `pkgs/top-level/python-packages.nix`.

Reply:
I did that in the first place but @pbsds and I discovered that building the binary together with the python package is a huge overhead in terms of build time. That's the reason for the split.

Aside from that a single file becomes hard to read because of the required dependencies very significantly based on whether the binary or the python module is built.

Or do you think we should go the route with the crowded file?

Reply:
Crowded files are quite common.A bunch of `++ lib.optionals enablePython [...]` could be the way to go.
 
The important difference is the make flags, which can be conditioned on `enablePython`.
Considering the base horizon-eda is now _not_ built when building the python module, the flag `enablePython` is IMO better named `buildPythonModule`, as it also disables the normal build.

Reply:
Okay, I'll change that.

(For me as a newbie those crowded files are super frustrating. It's hard to get started with nixpkgs. This PR already consumed days of work just because it's hard to figure out where things are coming from and how they work. And I'm guesstimating stuff till this day :D)

Reply:
I hope I understood your suggestion correctly? Let me know if this needs further adjustments.